client/src/components/expectations-list.tsx

```diff
@@ -50,8 +53,64 @@ const ExpectationCard = (props: {
     handleExpectationChange,
     handleRemoveExpectation,
     handleHintChange,
+    handlefeaturesChange,
   } = props;
+
   const [expanded, setExpanded] = React.useState(true);
+  const [curJson, setCurJson] = React.useState({});
+  const editorRef = React.useRef<any>();
+
+  const ajv = new Ajv({ allErrors: true, verbose: true });
+  /* eslint-disable-next-line @typescript-eslint/no-var-requires */
+  const schema = require("schemas/expectation-feature-schema.json");
+  let json = {};
+
+  React.useEffect(() => {
+    let expFeatures = {};
+    try {
+      if (expectation.features) {
+        expFeatures = JSON.parse(expectation.features);
+      } else {
+        expFeatures = { bad: [], good: [] };
+      }
+    } catch (e) {
+      console.error(e);
+      expFeatures = { bad: [], good: [] };
+    }
+    json = {
+      ...curJson,
+      ...expFeatures,
+      ideal: expectation.expectation,
+    };
+    try {
+      editorRef?.current.jsonEditor.set(json);
+    } catch (e) {
+      console.error(e);
+    }
+    setCurJson(json);
```

Comment:
The try catch here is mainly in case editorRef is null or doesn't have jsonEditor. For instance, when collapsing and uncollapsing the expectation the editorRef disappears for a bit.

SetCurJson(json) is saving the json changes the user made in the editor, to be reapplied when the expectation updates (otherwise they'd lose the changes they made after updating expectation text)